### ISSUE TITLE
DependencyResolver lost dependent info after plugin stop

### ISSUE THREAD

**Issue by hank-cp** [pf4j/pf4j#394]
Have two plugins A and B, B depends on A.

### Step to reproduce:
1. Resolve and start A/B
2. Stop A, A/B will be both `STOPPED`, which is expected.
3. Start A.
4. Start B,
5. Stop A again, A is `STOPPED` but B remains `STARTED`, which expected to be both `STOPPED`.

### Possible cause
in `AbstractPluginManager.stopPlugin(plugin, stopDependents)` methods, the following line
```
String dependent = dependents.remove(0);
```
also change the dependencyResolver, the dependent info is dropped with plugin stopping, unless we resolve everything again.

*a reproducible repo should be found [here](https://github.com/j-barata/marketbasket-sbp-sample). The pf4j version it's using is v3.1.0, and same behavior under v3.4.1*

**Comment by JoaoBrlt** [comment]
+1 The issue is clearly reproductible.

**Comment by decebals** [comment]
Do you have any solution already (idea or PR)?
@hank-cp Is the problem still relevant or did you find a workaround?

**Comment by j-barata** [comment]
As far as I know, there is no workaround. But this issue is still relevant (at least, for me).

**Comment by hank-cp** [comment]
@decebals I am not sure the purpose of dropping dependent info when unloading plugin. If it's ok to keep, simply change that line of code to should solve the problem.
```
    String dependent = dependents.get(0);
```

see #416. Not sure will this change break other place, please have a review on it.

**Comment by decebals** [comment]
A new SNAPSHOT version was created. Please check it. If the result is good I can release a new version.
Thanks!

**Comment by hank-cp** [comment]
@j-barata could you please verify this? I have tested on sbp and it's good.

**Comment by j-barata** [comment]
Hi @hank-cp and @decebals
Thank you both for your involvement in this issue resolution.
I will verify this. Is there any sbp-0.1.9-SNAPSHOT available including this fix ?

**Comment by decebals** [comment]
The fix is available in version 3.6.0 (released today).
